src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -220,6 +220,11 @@ public override void OnActionExecuted(ActionExecutedContext context)
                     }
                 }
             }
+            else if (context.Exception is ArgumentOutOfRangeException argumentOutOfRangeException)
```

Comment:
>ArgumentOutOfRangeException [](http://example.com/codeflow?start=42&length=27)

This is dangerous. 
It creates blank policy to treat any ArgumentOutOfRangeException which is base .net exception as Index exception.


In my opinoin we should throw BadRequestException in case of bad parameters and we already know how to work with it in this filter.

Reply:
BadRequestException makes sense.  Another possible option is a new Exception type which extends ArgumentOutOfRange.